src/org/nejrasm/zadaca2oop/homeworktask2/Robot.java

```diff
@@ -0,0 +1,82 @@
+package org.nejrasm.zadaca2oop.homeworktask2;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Robot {
+    final private int initialXPosition;
+    final private int initialYPosition;
+    private int currentXPosition;
+    private int currentYPosition;
+    private int movesCounter = 0;
+    private List<String> movesHistory = new ArrayList<>();
```

Comment:
The class works, but, you are storing moves as a list of `String`. Think about it logically if Bookshelf stores books, is there something you could do with Robot, to make it more clean, and avoid manipulating ints and strings.

Reply:
Done. I have added Moves class with positions and used that for moving Robot and filling the array list of movesHistory.